xilem/src/view/slider.rs

```diff
@@ -0,0 +1,265 @@
+    #[must_use]
+    /// Set the slider's direction (horizontal or vertical).
+    pub fn direction(mut self, axis: Axis) -> Self {
+        self.axis = axis;
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's thumb color.
+    pub fn with_color(mut self, color: impl Into<Color>) -> Self {
+        self.color = Some(color.into());
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's track color.
+    pub fn with_track_color(mut self, track_color: impl Into<Color>) -> Self {
+        self.track_color = Some(track_color.into());
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's step amount.
+    pub fn with_step(mut self, step: f64) -> Self {
+        self.step = Some(step);
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's thumb radii.
+    pub fn with_thumb_radii(mut self, radii: impl Into<RoundedRectRadii>) -> Self {
+        self.thumb_radii = Some(radii.into());
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's track radii.
+    pub fn with_track_radii(mut self, radii: impl Into<RoundedRectRadii>) -> Self {
+        self.track_radii = Some(radii.into());
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's hover glow color.
+    pub fn with_hover_glow_color(mut self, color: impl Into<Color>) -> Self {
+        self.hover_glow_color = Some(color.into());
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's hover glow blur radius.
+    pub fn with_hover_glow_blur_radius(mut self, blur_radius: f64) -> Self {
+        self.hover_glow_blur_radius = Some(blur_radius);
+        self
+    }
+
+    #[must_use]
+    /// Set the slider's hover glow spread radius.
+    pub fn with_hover_glow_spread_radius(mut self, spread_radius: f64) -> Self {
+        self.hover_glow_spread_radius = Some(spread_radius);
+        self
+    }
+
+    #[must_use]
+    /// Set the callback for editing state changes.
+    pub fn on_editing_changed(
+        mut self,
+        on_editing_changed: impl Fn(&mut State, bool) -> Action + Send + Sync + 'static,
+    ) -> Self {
+        self.on_editing_changed = Some(Box::new(on_editing_changed));
+        self
+    }
+}
+
+macro_rules! define_slider_properties {
+    ($($field:ident: $ty:ty => ($with_method:ident, $set_method:ident)),+ $(,)?) => {
+        macro_rules! build_slider {
+            ($slider:expr, $config:expr) => {{
+                let mut s = $slider;
+                $(
+                    if let Some(v) = $config.$field {
+                        s = s.$with_method(v);
+                    }
+                )*
+                s
+            }};
+        }
+
+        macro_rules! update_slider {
+            ($widget:expr, $current:expr, $prev:expr) => {
+                $(
+                    if $current.$field != $prev.$field {
+                        if let Some(v) = $current.$field {
+                            MasonrySlider::$set_method($widget, v);
+                        }
+                    }
```

Comment:
I'm not convinced that this behaviour is right.  I think it would fall down if `prev` is `Some(...)` and `current` is `None`.